src/main/java/org/jenkinsci/plugins/jiraext/view/FirstWordOfCommitStrategy.java

```diff
@@ -69,13 +69,23 @@ public FirstWordOfCommitStrategy()
         String msg = change.getMsg();
         String firstWordOfTicket;
         firstWordOfTicket = msg.substring(0, (msg.contains(" ") ? StringUtils.indexOf(msg, " ") : msg.length()));
+        firstWordOfTicket = firstWordOfTicket.substring(0, (firstWordOfTicket.contains("]") ? StringUtils.indexOf(firstWordOfTicket, "]") : firstWordOfTicket.length()));
+        firstWordOfTicket = firstWordOfTicket.substring(0, (firstWordOfTicket.contains(":") ? StringUtils.indexOf(firstWordOfTicket, ":") : firstWordOfTicket.length()));
 
         for (String validJiraPrefix : Config.getGlobalConfig().getJiraTickets())
         {
             if (firstWordOfTicket.endsWith(":"))
             {
                 firstWordOfTicket = firstWordOfTicket.substring(0, firstWordOfTicket.length() - 1);
             }
+            if (firstWordOfTicket.startsWith("["))
```

Comment:
This followed a bad example (line #77) -- this only supported removing a single character. Really what we want is to see if the first 'word' (string of characters without whitespace) contained a valid JIRA ticket. Tickets have the form `<prefix>-<number>` - we should be matching that to regex